api/main.py

```diff
@@ -44,20 +44,19 @@ def semantic():
 @app.route('/chat', methods=['POST'])
 @cross_origin()
 def chat():
-
     query = request.json.get('query')
     session_id = request.json.get('sessionId')
     history = request.json.get('history', [])
-    settings = Settings(**request.json.get('settings', {}))
-
-    def run(callback):
-        return run_query(session_id, query, history, settings, callback)
+    settings = request.json.get('settings', {})
 
     def formatter(item):
         if isinstance(item, Exception):
             item = {'state': 'error', 'error': str(item)}
         return json.dumps(item)
 
+    def run(callback):
+        return run_query(session_id, query, history, Settings(**settings), callback)
```

Comment:
the `Settings` class have various validators that can raise exceptions. This causes the request to fail with a uninformative (for the user) error, so make sure that a proper error instance is streamed back